### DISCUSSION TITLE
GraphQL schema retrieval with URL and authorization header

### DISCUSSION THREAD

**Discussion by palmamartin** [0no-co/gql.tada#54] [Q&A] [answered]
What's the best way to handle the case where an authorization token is needed to retrieve the GraphQL schema? 

```
{
        "name": "@0no-co/graphqlsp",
        "schema": {
          "url": "https://someendpoint",
          "headers": {
            "Authorization": "sometoken",
            "X-Exclude-Invalid": "true"
          }
        },
        "tadaOutputLocation": "./src/lib/graphql-env.d.ts"
      }
```

I do not prefer to store the token inside the `tsconfig.json` and commit that to a repository. Is there a way use environment variables inside in `tsconfig.json`?

**Comment by valtism** [comment]
I would like to be pulling the latest schema from our staging, but to do so I need to include an authorisation header in the schema config. 

Since this definition lives in `tsconfig.json`, the bearer token for auth has to be added in directly to the file.

I would like to be able to define this config elsewhere in a `.js`/`.ts` file so I can use `dotenv` to pull in the token from an env var. I have been previously doing this when using GraphQL Codegen.

**Comment by anirudhsama** [comment, marked as answer]
This is now possible with the cli. Configure `tsconfig.json` with a schema option like so:
```json
"plugins": [
	{
		"name": "@0no-co/graphqlsp",
		"tadaOutputLocation": "./graphql-env.d.ts",
		"schema": "./graphql/schema.graphql"
	}
]
```
And run the CLI command like below. It will output the schema file to the location specified above.

`gql-tada generate-schema $GRAPHQL_URL --header "authorization: $AUTH_TOKEN"`
